metrics/prometheus.go

```diff
@@ -454,6 +455,38 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 			},
 		}...)
 	}
+	if includedMetrics.Has(container.MemoryNumaMetrics) {
+		c.containerMetrics = append(c.containerMetrics, []containerMetric{
+			{
+				name:        "container_memory_numa_pages",
+				help:        "Memory usage per numa node",
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{"type", "scope", "node"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					values := make(metricValues, 0)
+
+					values = append(values, getNumaStatsPerNode(s.Memory.ContainerData.NumaStats.Total,
+						[]string{"total", "container"}, s.Timestamp)...)
+					values = append(values, getNumaStatsPerNode(s.Memory.ContainerData.NumaStats.File,
+						[]string{"file", "container"}, s.Timestamp)...)
+					values = append(values, getNumaStatsPerNode(s.Memory.ContainerData.NumaStats.Anon,
+						[]string{"anon", "container"}, s.Timestamp)...)
+					values = append(values, getNumaStatsPerNode(s.Memory.ContainerData.NumaStats.Unevictable,
+						[]string{"unevictable", "container"}, s.Timestamp)...)
+
+					values = append(values, getNumaStatsPerNode(s.Memory.HierarchicalData.NumaStats.Total,
+						[]string{"total", "hierarchy"}, s.Timestamp)...)
```

Comment:
I suspect hierarchy vs container may need to be separate metrics by the same logic above.

Reply:
I followed the pattern which was used for "container_memory_failures_total metric", see [this](https://github.com/google/cadvisor/blob/master/metrics/prometheus.go#L426). There is "scope" label with values "container" or "hierarchy".